debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -291,14 +290,18 @@ private PGReplicationStream startPgReplicationStream(final LogSequenceNumber lsn
     public synchronized void close() {
         try {
             super.close();
-        } catch (SQLException e) {
+        } catch (Throwable e) {
             LOGGER.error("Unexpected error while closing Postgres connection", e);
         }
-        if (dropSlotOnClose) {
-            // we're dropping the replication slot via a regular - i.e. not a replication - connection
-            try (PostgresConnection connection = new PostgresConnection(originalConfig)) {
-                connection.dropReplicationSlot(slotName);
+        try {
```

Comment:
Seems simpler to add the catch to the inner try(-with-resources)?

Reply:
@gunnarmorling Right!